ush/retrieve_data.py

```diff
@@ -289,6 +290,14 @@ def get_file_templates(cla, known_data_info, data_store, use_cla_tmpl=False):
     """
 
     file_templates = known_data_info.get(data_store, {}).get("file_names")
+    file_templates = deepcopy(file_templates)
+
+    # Remove sfc files from fcst in file_names of FV3GFS for LBCs
+    # sfc files needed in fcst when time_offset is not zero.
+    if cla.external_model == "FV3GFS" and cla.ics_or_lbcs == "LBCS":
```

Comment:
Can this just be if "lbcs" and "fcst"? That way we don't tightly couple the tool to specifics of the models we use as inputs?

Reply:
@christinaholtNOAA, I am afraid that it will cause another unexpected errors with other models. For example, 'GDAS', 'GSMGFS', 'RAP', 'HRRR', and 'NAM' have only one file name for 'fcst'. In these cases, we will have the same index error. For 'GEFS', we should not remove the second array from the file names. We just want to remove the 'sfc' file from the list. 'FV3GFS' only has this issue. What do you think of this?

Reply:
This logic is still pretty brittle. It assumes that there must be at least 2 and that the 2nd one is the surface file. I think that if we want to protect that assumption, we need a functional test. 

I also think it's probably safer to do something like delete the entry if "sfc" is in the name of the file. That type of logic may get us the additional functionality we'd need to use offset GDAS netcdf/nemsio as ICS and expand the logic to "lbcs" and "fcst" instead of FV3GFS-specific logic. In general that says "if we're looking for lbcs from forecasts and a surface file is listed, don't try to get it".

As best I can tell, sfc files only show up for GDAS, FV3GFS, and GSMGFS. 

I'd prefer to see something like this:

```
if cla.ics_or_lbcs == "LBCS":
    for format in ['netcdf', 'nemsio']:
        for i, tmpl in enumerate(file_templates.get('format', {}).get('fcst', [])):
            if "sfc" in tmpl:
                 del file_templates[format]['fcst'][i]
```

It reduces the assumptions about order and the length of the list provided. It assures that we're only removing surface files, and not atm files accidentally. It also helps us if we want to expand our experiments run with offset GDAS files as ICs -- we can just add the sfc file to the GDAS fcst entry.

Reply:
@christinaholtNOAA, I tested your script, but sfc file was not removed from the list for LBCs: `DEBUG: Looking for files like ['gfs.t{hh}z.atmf{fcst_hr:03d}.nemsio', 'gfs.t{hh}z.sfcf{fcst_hr:03d}.nemsio']`. Any suggestion?

Reply:
I found a typo there: ` 'format'` => `format`. I am testing it again.

Reply:
Apologies...I was just writing directly in the text box here and didn't do a full test.